packages/dht/src/connection/Handshaker.ts

```diff
@@ -101,4 +101,8 @@ export class Handshaker extends EventEmitter<HandshakerEvents> {
         this.connection.send(Message.toBinary(msg))
         logger.trace('handshake response sent')
     }
+
+    public stop(): void {
+        this.connection.off('data', (data: Uint8Array) => this.onData(data))
```

Comment:
Shouldn't the 2nd argument here be a reference to the same function that was assigned to listen to the event originally. E.g.
  ```suggestion
        this.connection.off('data', this.onDataListener)
```

And in constructor:
```ts
constructor(...) {
    // ...
    this.onDataListener = (data: Uint8Array) => this.onData(data)
    this.connection.on('data', this.onDataListener)
    // ...
}
```

Reply:
right... somehow I keep forgeting